core/trino-main/src/test/java/io/trino/operator/TestGroupByHash.java

```diff
@@ -502,4 +505,119 @@ public void testMemoryReservationYieldWithDictionary()
         assertEquals(currentQuota.get(), 10);
         assertEquals(currentQuota.get() / 3, yields);
     }
+
+    @Test
+    public void testLowCardinalityDictionariesAddPage()
+    {
+        GroupByHash groupByHash = createGroupByHash(
+                TEST_SESSION,
+                ImmutableList.of(BIGINT, BIGINT),
+                new int[] {0, 1},
+                Optional.empty(),
+                100,
+                JOIN_COMPILER,
+                TYPE_OPERATOR_FACTORY,
+                NOOP);
+        Block firstBlock = BlockAssertions.createLongDictionaryBlock(0, 100, 10);
+        Block secondBlock = BlockAssertions.createLongDictionaryBlock(0, 100, 10);
+        Page page = new Page(firstBlock, secondBlock);
+
+        groupByHash.addPage(page).process();
+        assertThat(groupByHash.getGroupCount()).isEqualTo(10); // Blocks are identical so only 10 distinct groups
+
+        firstBlock = BlockAssertions.createLongDictionaryBlock(10, 100, 5);
+        secondBlock = BlockAssertions.createLongDictionaryBlock(10, 100, 7);
+        page = new Page(firstBlock, secondBlock);
+
+        groupByHash.addPage(page).process();
+        assertThat(groupByHash.getGroupCount()).isEqualTo(45); // Old 10 groups and 35 new
+    }
+
+    @Test
+    public void testCompareLowCardinalityWithGetGroupIds()
+    {
+        GroupByHash groupByHash = createGroupByHash(
```

Comment:
add a comment like: `compare group id results from page with dictionaries only (processed via low cardinality work) vs page that is processed normally.`